File: Core.DomainModel/ItSystem/ItSystemBase.cs
using System;

namespace Core.DomainModel.ItSystem
{
    public abstract class ItSystemBase : Entity, ISystemModule, IOwnedByOrganization, IHasAccessModifier, IHasName, IHasUuid
    {
        protected ItSystemBase()
        {
            Uuid = Guid.NewGuid();
        }

        public string Name { get; set; }
        public Guid Uuid { get; set; }
        public string Description { get; set; }
        
        public AccessModifier AccessModifier { get; set; }

        public int OrganizationId { get; set; }
        /// <summary>
        /// Gets or sets the organization this instance was created under.
        /// </summary>
        /// <value>
        /// The organization.
        /// </value>
        public virtual Organization.Organization Organization { get; set; }

        public DateTime? Created { get; set; }
    }
}

File: Presentation.Web/app/models/it-system/it-system-base.ts
module Kitos.Models.ItSystem {
    export interface IItSystemBase extends IEntity {
        Name: string;
        Uuid: any;
        Description: string;
        Url: string;
        AccessModifier: AccessModifier;
        OrganizationId: number;
        /** Gets or sets the organization this instance was created under. */
        Organization: IOrganization;
        BelongsToId: number;
        /** Gets or sets the organization the system belongs to. */
        BelongsTo: IOrganization;
    }
}
